url21b471891f_Agetac_Model_tgz-pJ8-org_json_TestJ8/src/org/agetac/common/api/InterventionConnection.java

```diff
@@ -0,0 +1,651 @@
+			for (int i = 0; i < ar.length(); i++) {
+				messages.add(new Message(ar.getJSONObject(i)));
+			}
+
+		} catch (IOException e) {
+			e.printStackTrace();
+		} catch (JSONException e) {
+			e.printStackTrace();
+		}
+
+		return messages;
+	}
+	@Override
+	public Message putMessage(Message msg) throws JSONException, BadResponseException {
+
+		Representation r = new JsonRepresentation(msg.toJSON());
+		r = serv.putResource("intervention/" + interId + "/message", null, r);
+		Message message = null;
+		try {
+			message = new Message(new JsonRepresentation(r).getJsonObject());
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+
+		return message;
+	}
+	@Override
+	public Message postMessage(Message msg) throws JSONException, BadResponseException {
+
+		Representation r = new JsonRepresentation(msg.toJSON());
+
+		r = serv.postResource("intervention/" + interId + "/message", msg.getUniqueID(), r);
+		Message message = null;
+		try {
+			 message =  new Message(new JsonRepresentation(r).getJsonObject());
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+		return  message;
+
+	}
+	@Override
+	public void deleteMessage(Message msg) throws BadResponseException {
+		serv.deleteResource("intervention/" + interId + "/message",	msg.getUniqueID());
+	}
+	
+	@Override
+	public Vehicule getVehicule(String vId) throws BadResponseException {
+		Vehicule v = null;
+		JsonRepresentation representation = null;
+
+		Representation repr = serv.getResource("intervention/" + interId
+				+ "/vehicule", vId);
+
+		try {
+			representation = new JsonRepresentation(repr);
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+
+		try {
+			v = new Vehicule(representation.getJsonObject());
+		} catch (InvalidJSONException e) {
+			e.printStackTrace();
+		} catch (JSONException e) {
+			e.printStackTrace();
+		}
+
+		return v;
+	}
+	@Override
+	public List<Vehicule> getVehicules() throws BadResponseException {
+
+		List<Vehicule> vehicules = new ArrayList<Vehicule>();
+		JsonRepresentation representation = null;
+
+		// Rďż˝cupďż˝ration de la liste des Vehicules
+		Representation repr = serv.getResource("intervention/" + interId
+				+ "/vehicule", null);
+
+		try {
+			representation = new JsonRepresentation(repr);
+			JSONArray ar = representation.getJsonArray(); 
+			for (int i = 0; i < ar.length(); i++) {
+				vehicules.add(new Vehicule(ar.getJSONObject(i)));
+			}
+
+		} catch (Exception e) {
+			System.out.println("Error: " + e.toString());
+		}
+
+		return vehicules;
+	}
+	@Override
+	public Vehicule putVehicule(Vehicule v) throws BadResponseException, JSONException {
+
+		Representation r = new JsonRepresentation(v.toJSON());
+
+		r = serv.putResource("intervention/" + interId + "/vehicule", null, r);
+		Vehicule vehicule = null;
+		try {
+			vehicule = new Vehicule(new JsonRepresentation(r).getJsonObject());
+		} catch (IOException e) {
+			e.printStackTrace();
+		} catch (InvalidJSONException e) {
+			e.printStackTrace();
+		}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 186:192, 207:213, 272:278, 290:296, 338:344, 383:389, 511:517, 557:563, 595:601, 640:646.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.